mautic/mautic
# Email A/B test fixes

**Description**
1) The big thing with this PR is that it fixes an issue where single emails sent via campaigns, etc never used a variant regardless of the weight assigned to it due to the EmailModel::sendEmail function calculating the weights and limits, but not ignoring an email with a calculated weight of 0. 

2) For batch emails, such as list sends, one extra email was sent for each variant email due to the use of > rather than >=.

3) It also fixes an issue where the A/B winning criteria, "Read Rate," compared the a/b test start date with the read_date of the stat rather than the sent_date skewing the results.

4) Finally, the stats displayed in the list were of the parent only making it confusing as to the real send/read counts overall.  So this PR ensures that the sent/read counts in the Manage Emails lists include a summation of all variants.  It also adjusted the graph to give an option to view the counts for the specific variant or for all variants.  It defaults to "all" for the "parent" and "variant" for "children."

**Testing**
Create an template email with at least one A/B variant of 50% and a read rate winning criteria.  Then create a campaign to send the email and assign at least two leads to the campaign. Run the command `mautic:campaign:trigger.` The emails should be sent out to the leads. Do not open the emails sent to the leads.

Things to note:
1) Browse to the email's details page. All emails were sent to the one email when it should have been divided 50/50 with the variant. (# 1 under Description)

2) Click the Variants tab then A/B Test Stats. Notice that the sent counts are 0 when they should be the total number sent. (# 3  under Description)

Now create a new list based email, assign it to a list with at 4 leads, and then create an A/B variant of 50% and a read rate winning criteria.  Send the email to the list.

Things to note:
1) Look at the stats in the Manage Email list for the email just created. It'll only show the stats for the parent email and does not include those sent to the variants. (# 4  under Description)

2) One extra email is sent to the parent email. So 3 was sent to the parent and 1 to the variant when it should be 2 and 2. (# 2 under Description)

Apply the PR and repeat the process.  It's best and easiest to create new emails.
1) The stats in the Manage Email list will now include a summation of all variants.
2) The emails sent through the campaign should be properly distributed 50/50
3) The emails sent via a  list should be properly distributed 50/50
4) The A/B Test Stats graph should now have a proper sent count
5) The summary graphs of each template/list email should default to include stats for all variants. Click on the Variant link above upper right corner of the graph and it will show stats just for the specific variant.  (Defaults to "all" for the parent but if you click on a variant from the Variants tab, it'll default to just the variant's stats).
